### PR TITLE
Clientpr/import schema

### PR DESCRIPTION
**Pull request by sivaRajanaDivami**
# TITLE

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

### CODE REVIEW

File: app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx [hunk 1]
```diff
@@ -69,6 +70,15 @@ export default function ConnectionsTabs({
     setConnectionId(connectionId);
   }, []);
 
+  if (index === 0) {
```
Comment (tharunpola) [inline app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx]: convert it to index === 0 && (DataPrepStore......)

File: app/cdap/components/GridTable/index.tsx [hunk 2]
```diff
@@ -50,12 +52,20 @@ export default function GridTable() {
   const [rowsDataList, setRowsDataList] = useState([]);
   const [gridData, setGridData] = useState({} as IExecuteAPIResponse);
   const [missingDataList, setMissingDataList] = useState([]);
-  const [invalidCountArray, setInvalidCountArray] = useState([
+  const [invalidCount, setInvalidCount] = useState<IInvalidCount[]>([
```
Comment (tharunpola) [inline app/cdap/components/GridTable/index.tsx]: rename it to [invalidDataCount, setInvalidDataCount]

File: app/cdap/components/ParsingDrawer/Components/InputSelect/index.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      defaultValue={defaultValue}
+      MenuProps={{
+        anchorOrigin: {
+          vertical: 'bottom',
+          horizontal: 'left',
+        },
+        getContentAnchorEl: null,
+      }}
+    >
+      {Array.isArray(options) &&
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/Components/InputSelect/index.tsx]: options type as an array is already defined in the interface, and Array.isArray(options) is not needed

File: app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/__test__/ParsingHeaderActionTemplate.test.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render } from '@testing-library/react';
+import { createBrowserHistory as createHistory } from 'history';
+import React from 'react';
+import { Route, Router, Switch } from 'react-router';
+import ParsingHeaderActionTemplate from '..';
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/__test__/ParsingHeaderActionTemplate.test.tsx]: please add full path

File: app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/index.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    };
+  };
+  return (
+    <Box className={classes.importSchemaIconText}>
+      <input
+        data-testId="fileinput"
+        id="file"
+        type="file"
+        accept=".json"
+        style={{ display: 'none' }}
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/index.tsx]: please remove inline styles

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+export default function({ values, changeEventListener }: IParsingPopupBodyProps) {
+  const classes = useStyles();
+
+  const { format, fileEncoding, enableQuotedValues, skipHeader } = values;
+  let selectedFormatValue: IOptions[] = [];
+  let selectedEncodingValue: IOptions[] = [];
+
+  useEffect(() => {
+    selectedFormatValue = FORMAT_OPTIONS?.filter((i) => i.value === format);
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx]: replace i with eachOption

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+import T from 'i18n-react';
+
+const PREFIX = 'features.DataPrep.Directives.SetCharEncoding';
+const SUFFIX = 'features.DataPrep.Directives.Parse';
+
+export const CHAR_ENCODING_OPTIONS = [
+  {
+    label: T.translate(`${PREFIX}.utf8`),
+    value: T.translate(`${PREFIX}.utf8`),
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts]: the label can be stored in text-en.yaml
but values are just constants and it is unnecessary to be placed in text-en.yaml file

File: app/cdap/components/ParsingDrawer/index.tsx [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';
+import { createWorkspace } from 'components/Connections/Browser/GenericBrowser/apiHelpers';
+import { ConnectionsContext } from 'components/Connections/ConnectionsContext';
+import DataPrepStore from 'components/DataPrep/store';
+import DrawerWidget from 'components/DrawerWidget';
+import ParsingHeaderActionTemplate from './Components/ParsingHeaderActionTemplate';
+import PositionedSnackbar from 'components/SnackbarComponent/index';
+import T from 'i18n-react';
+import React, { useContext, useEffect, useState } from 'react';
+import ParsingPopupBody from './Components/ParsingPopupBody';
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/index.tsx]: update it to the full path

File: app/cdap/components/ParsingDrawer/index.tsx [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+import {
+  defaultConnectionPayload,
+  defaultErrorOnTransformations,
+  defaultProperties,
+} from './defaultValues';
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/index.tsx]: update it to the full path

File: app/cdap/components/ParsingDrawer/index.tsx [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+import { useStyles } from './styles';
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/index.tsx]: update it to the full path

File: app/cdap/components/ParsingDrawer/index.tsx [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+import Alert from '@material-ui/lab/Alert';
+
+export default function({ setLoading, updateDataTranformation }) {
+  const [drawerStatus, setDrawerStatus] = useState(true);
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/index.tsx]: define types to all useStates if possible

File: app/cdap/components/ParsingDrawer/index.tsx [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+  const [successUpload, setSuccessUpload] = useState({ open: false, message: '' });
+  const [connectionPayload, setConnectionPayload] = useState(defaultConnectionPayload);
+  const [failureSchema, setFailureSchemaStatus] = useState(false);
+  const [toaster, setToaster] = useState({ lastValue: null });
+
+  const classes = useStyles();
+  const { dataprep } = DataPrepStore.getState();
+
+  useEffect(() => {
+    if (failureSchema && toaster.lastValue !== 'fail') {
```
Comment (tharunpola) [inline app/cdap/components/ParsingDrawer/index.tsx:63]: remove nested if else

File: app/cdap/components/SnackbarComponent/Components/TransitionComponent/index.tsx [hunk 13]
```diff
@@ -18,27 +18,34 @@ import React from 'react';
 import WarningRoundedIcon from '@material-ui/icons/WarningRounded';
 import { Divider } from '@material-ui/core';
 import { useStyles } from './styles';
+import CheckCircleIcon from '@material-ui/icons/CheckCircle';
+import { ITransitionComponentProps } from './types';
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/Components/TransitionComponent/index.tsx]: add full import path

File: app/cdap/components/SnackbarComponent/Components/TransitionComponent/index.tsx [hunk 14]
```diff
@@ -18,27 +18,34 @@ import React from 'react';
...
+  const messenger =
+    snackbarStatus === 'success' ? (
+      <h5 className={classes.successHead}>
+        <CheckCircleIcon fontSize="large" className={classes.successIcon} />
+        &nbsp; Step Successfully added!
+      </h5>
+    ) : (
+      <h5 className={classes.errorHead}>
+        <WarningRoundedIcon className={classes.warningIcon} />
+        &nbsp;Error
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/Components/TransitionComponent/index.tsx]: remove &nbsp and try to add padding and margin

File: app/cdap/components/SnackbarComponent/Components/TransitionComponent/styles.ts [hunk 15]
```diff
@@ -14,10 +14,15 @@
...
 
 export const useStyles = makeStyles((theme) => ({
+  undoSpan: {
+    display: 'block',
+    fontSize: '14px',
+    cursor: 'pointer',
+    color: green[900],
+  },
   warningIcon: {
     color: '#E97567',
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/Components/TransitionComponent/styles.ts]: check if colors are available in MUI color pallet

File: app/cdap/components/SnackbarComponent/index.tsx [hunk 16]
```diff
@@ -17,8 +17,13 @@ import React, { useState, useEffect } from 'react';
 import Snackbar from '@material-ui/core/Snackbar';
 import { useStyles } from './styles';
 import TransitionComponent from './Components/TransitionComponent';
+import { IPositionedSnackbarProps } from './types';
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/index.tsx]: add full import path

File: app/cdap/components/SnackbarComponent/index.tsx [hunk 17]
```diff
@@ -35,16 +40,24 @@ export default function PositionedSnackbar(props) {
...
   return (
     <Snackbar
       anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
       open={isOpen}
       onClose={handleClose}
-      TransitionComponent={() => <TransitionComponent close={() => handleClose()} />}
-      className={classes.snackBarDiv}
+      TransitionComponent={() => (
+        <TransitionComponent
+          close={() => handleClose()}
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/index.tsx]: replace above line with close={ handleClose}

File: app/cdap/components/SnackbarComponent/styles.ts [hunk 18]
```diff
@@ -15,9 +15,20 @@
...
+import { green } from '@material-ui/core/colors';
 
 export const useStyles = makeStyles({
-  snackBarDiv: {
+  snackBarDivSuccess: {
+    padding: '10px',
+    display: 'block',
+    border: '1px solid #4BAF4F',
+    boxShadow: '-3px 4px 15px rgba(68, 132, 245, 0.25)',
+    height: '148px',
```
Comment (tharunpola) [inline app/cdap/components/SnackbarComponent/styles.ts]: convert px to numbers all over the file

File: app/cdap/text/text-en.yaml [hunk 19]
```diff
@@ -3431,9 +3431,26 @@ features:
         title: Configure a pipeline
       success: You have successfully created the pipeline "{pipelineName}".
```
Comment (tharunpola) [inline app/cdap/text/text-en.yaml:3432]: check all the new changes in alphabetical order

**Review by tharunpola — CHANGES_REQUESTED** [review]

File: app/cdap/components/GridTable/index.tsx (hunk 2 above)
Comment (dlakshmi123) [reply app/cdap/components/GridTable/index.tsx]: done

File: app/cdap/components/ParsingDrawer/Components/InputSelect/index.tsx (hunk 3 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/Components/InputSelect/index.tsx]: done

File: app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/__test__/ParsingHeaderActionTemplate.test.tsx (hunk 4 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/__test__/ParsingHeaderActionTemplate.test.tsx]: done

File: app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/index.tsx (hunk 5 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/Components/ParsingHeaderActionTemplate/index.tsx]: done

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx (hunk 6 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/index.tsx]: done

File: app/cdap/components/ParsingDrawer/index.tsx (hunk 8 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/index.tsx]: done

File: app/cdap/components/ParsingDrawer/index.tsx (hunk 9 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/index.tsx]: done

File: app/cdap/components/ParsingDrawer/index.tsx (hunk 10 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/index.tsx]: done

File: app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts (hunk 7 above)
Comment (dlakshmi123) [reply app/cdap/components/ParsingDrawer/Components/ParsingPopupBody/parsingOptions.ts]: done

File: app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx (hunk 1 above)
Comment (dlakshmi123) [reply app/cdap/components/ConnectionList/Components/ConnectionTabs/index.tsx]: done

**Pull request closed without merging**